src/pres/MainContainer.js

```diff
@@ -38,28 +39,28 @@ export default class MainContainer extends React.Component {
   constructor(props){
     super(props)
 
-    let urlVariant = new URLSearchParams(window.location.search).get("variant")
-    let selectedVariant = urlVariant || Constants.VARIANT_STANDARD
-    this.chess = chessLogic(selectedVariant)
+    const urlSettings = this.getSettingsFromUrlParams()
+    this.chess = chessLogic(urlSettings.variant)
     addStateManagement(this)
     this.state = {
         resize:0,
         fen: this.chess.fen(),
         lastMove: null,
         gamesProcessed:0,
-        openingGraph:new OpeningGraph(selectedVariant),
+        openingGraph:new OpeningGraph(urlSettings.variant),
         settings:{
           playerName:'',
           orientation:Constants.PLAYER_COLOR_WHITE,
           playerColor:'',
           movesSettings:this.getMovesSettingsFromCookie(),
           darkMode: this.getDarkModeSettingFromCookie()
         },
+        urlSettings,
```

Comment:
Something I'm not sure about in React:

`urlSettings` should never be mutated (at least, as the code is right now). So I'm not sure if this belongs in `state`! Is there somewhere else that this sort of thing goes?